### ISSUE TITLE
Prefer Lodash get

### ISSUE THREAD

**Issue by GLosch** [wework/eslint-config-wework#4]
## The problem

In code reviews, a common request is to use Lodash `_.get` in place of referencing deeply nested data structures like `foo.bar.moreFoo.moreBar` to avoid potential JavaScript errors that would break the page. Implementing the `prefer-get` rule [here](https://github.com/wix/eslint-plugin-lodash/blob/master/docs/rules/prefer-get.md) would take care of this.

**Comment by rfarine** [comment]
No qualms with this at all. Sounds good to me. :+1:

**Comment by kangax** [comment]
How deep is too deep or would that be determined on a per-project basis?

**Comment by GLosch** [comment]
default depth is 3, which sounds like a reasonable baseline to me

**Comment by kangax** [comment]
I'm generally for it although to play devil's advocate:

1. deep nesting could be an indication of a code smell (code reaching "far beyond" its concerns/responsibilities; law of Demeter)
2. if the errors are swallowed silently, it's not always a good thing

Having said that, there's definitely valid cases for deep access (e.g. retrieving something from a dumb but deeply nested data structure) where `get` could be useful.

One more thing — in our codebase I see quite a few occurrences of triple nesting such as
 `this.props.fetchChurnItems(...)`, `state.churnItems.loading`, `props.currentLocation.uuid`, and so on. I wouldn't want any of these wrapped in `_.get`. It feels like I'd want to use `_.get` in very specific cases where there's 4+ nesting and only on a pure data structure.

**Comment by GLosch** [comment]
Yeah, great points, @kangax. For some added context, the use case that brought this up in the first place was our Contentful entries. A common resource like a hero image url could come back nested as follows: `fields.section.fields.background.fields.file.url`. 

...and that's following normal practices for Contentful data modeling. 

Error handling/alerting is a different question though, and probably something better left to the application engineers to determine.

**Comment by GLosch** [comment]
re: that thread in [quincy-2337](https://github.com/WeConnect/www-quincy/issues/2337), it would be great if this lint config also enforced importing lodash as `_` and using functions prefixed with `_`, e.g. `_.get()` instead of importing the named function and using it like `get()`

**Comment by nason** [comment]
Thats a slightly different conversation @GLosch -- this one is talking about preferring get for lookups in deeply nested objects.

What you're talking about is importing the whole lodash module vs importing specific modules as necessary. I disagree with you on this one -- let's open up a new issue to discuss?

**Comment by nason** [comment]
Opened https://github.com/wework/eslint-config-wework/issues/5 for that discussion ^ /cc @GLosch @thinkaxelthink

**Comment by GLosch** [comment]
awesome, thanks
